alacritty/src/event.rs

```diff
@@ -434,6 +437,16 @@ impl<N: Notify + OnResize> Processor<N> {
                 );
             }
 
+            // Check for URL below cursor
+            let highlighted_url = self.highlighted_url(&terminal);
+
+            // Check for URL highlight change
+            let prev_highlighted = mem::replace(&mut url_highlighted, highlighted_url.is_some());
+            terminal.dirty |= prev_highlighted != url_highlighted;
+
+            // Update mouse cursor shape
+            self.display.window.set_mouse_cursor(self.mouse_cursor(&terminal, url_highlighted));
```

Comment:
Does it actually improves performance, since you're performing URL search/highlighting inside the terminal lock stage?

Reply:
*Improving* performance is not the goal of this PR, but I haven't sufficiently tested it yet. There shouldn't be a measurable performance difference.

Reply:
Then could you elaborate, what’s the goal of this PR?

Reply:
It significantly simplifies the URL matching and removes logic that just isn't necessary for the URLs we query.

It manages to improve our URL matching results, since it allows highlighting URLs regardless of what is visible and at the same time it removes like ~250 lines of code.

As far as I'm concerned, this can do the same thing (and more) our current URL matcher can do, while using simpler code to do it. However of course this approach might be less relevant if we would want to always underline all URLs for example.